mmv1/third_party/terraform/services/privilegedaccessmanager/data_source_privileged_access_manager_entitlement_test.go

```diff
@@ -0,0 +1,120 @@
+package privilegedaccessmanager_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccDataSourceGooglePrivilegedAccessManagerEntitlement_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckPrivilegedAccessManagerEntitlementDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGooglePrivilegedAccessManagerEntitlement_basic(context),
+				Check: resource.ComposeTestCheckFunc(
+					acctest.CheckDataSourceStateMatchesResourceState("data.google_privileged_access_manager_entitlement.tfentitlement", "google_privileged_access_manager_entitlement.tfentitlement"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceGooglePrivilegedAccessManagerEntitlement_optionalProject(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckPrivilegedAccessManagerEntitlementDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGooglePrivilegedAccessManagerEntitlement_optionalProject(context),
+				Check: resource.ComposeTestCheckFunc(
+					acctest.CheckDataSourceStateMatchesResourceState("data.google_privileged_access_manager_entitlement.tfentitlement", "google_privileged_access_manager_entitlement.tfentitlement"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceGooglePrivilegedAccessManagerEntitlement_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_privileged_access_manager_entitlement" "tfentitlement" {
+	entitlement_id = "tf-test-example-entitlement%{random_suffix}"
+	location = "global"
+	max_request_duration = "43200s"
+	parent = "projects/%{project_name}"
```

Comment:
`project_name` is missing from the context map in both of these test configurations. If the data source doesn't default to a project id for the parent, you'll need to use a `google_project` datasource to get the test project's project_id and interpolate it here. (Search for `data "google_project"` in the codebase for examples.)